presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/BasePickTableLayoutTest.java

```diff
@@ -56,7 +57,7 @@ public BasePickTableLayoutTest(boolean predicatePushDownEnabled)
     @BeforeMethod
     public void setUpPerMethod()
     {
-        pickTableLayout = new PickTableLayout(tester().getMetadata());
+        pickTableLayout = new PickTableLayout(tester().getMetadata(), new SqlParser());
```

Comment:
Would it be possible to added a test case in `TestPickTableLayoutWithPredicatePushDown`? I understand this needs to be a condition involving `orderstatus` (a column that acts as a partitioning column) that filters certain values  but is not convertible into `TupleDomain`, right? Like `codepoint(orderstatus) < random()` ?

Reply:
That way you will get table layout with all partitions in `currentConstraint`, that test is good.  I think the other which will actually `shouldPrune` would a expression `strpos(orderstatus, '0') = 1`